version/src/lib.rs

```diff
@@ -198,6 +201,12 @@ pub mod scale {
         /// # Errors
         /// Will return error if version is unsupported or if input won't have enough bytes for decoding.
         fn decode_versioned(input: &[u8]) -> Result<Self>;
+
+        /// Use this function for versioned objects instead of `decode_all`.
+        ///
+        /// # Errors
+        /// Will return error if version is unsupported or if input won't have enough bytes for decoding.
```

Comment:
I'd suggest making this a list:

```suggestion
        /// Return error if:
        /// - version is not supported
        /// - input doesn't have enough bytes for decoding
```